shared/src/ledger/storage/merkle_tree.rs

```diff
@@ -56,7 +56,7 @@ pub enum Error {
 type Result<T> = std::result::Result<T, Error>;
 
 /// Type alias for bytes to be put into the Merkle storage
-pub(super) type StorageBytes = Vec<u8>;
+pub(super) type StorageBytes<'a> = &'a [u8];
```

Comment:
since values of this type can either be owned or borrowed, it could also be changed to:
```suggestion
pub(super) type StorageBytes<'a> = std::borrow::Cow<'a, [u8]>;
```

Reply:
So I implemented this, and I got a lot of added verbosity and I'm not sure what I won.

Reply:
hmmm. not all battles can be won. sorry xD

Reply:
Well, let me fiddle with this a bit longer

Reply:
Yeah...., to get a a benefit from this is major refactor. And the benefit is that I save a clone in a single method. Just doesn't seem worth it at the moment.